arcade/camera/camera_2d.py

```diff
@@ -0,0 +1,713 @@
+
+        Args:
+            window: The Arcade Window to bind the camera to.
+                    Defaults to the currently active window.
+            viewport: A 4-int tuple which defines the pixel bounds which the camera with project to.
+            position: The 2D position of the camera in the XY plane.
+            up: The 2D unit vector which defines the +Y-axis of the camera space.
+            zoom: A scalar value which is inversely proportional to the size of the camera projection.
+                  i.e. a zoom of 2.0 halves the size of the projection, doubling the perceived size of objects.
+            projection: A 4-float tuple which defines the world space
+                        bounds which the camera projects to the viewport.
+            near: The near clipping plane of the camera.
+            far: The far clipping place of the camera.
+            camera_data: A CameraData PoD which describes the viewport, position, up, and zoom
+            projection_data: A OrthographicProjectionData PoD which describes the left, right, top,
+                             bottom, far, near planes for an orthographic projection.
+        """
+        self._window: "Window" = window or get_window()
+
+        assert (
+            not any((viewport, position, up, zoom)) or not bool(camera_data)
+        ), (
+            "Camera2D Warning: Provided both a CameraData object and raw values. Defaulting to CameraData."
+        )
+
+        assert (
+            not any((projection, near, far)) or not bool(projection_data)
+        ), (
+            "Camera2D Warning: Provided both an OrthographicProjectionData object and raw values."
+            "Defaulting to OrthographicProjectionData."
+        )
+
+        _pos = position or (self._window.width / 2, self._window.height / 2)
+        _up = up or (0.0, 1.0)
+        self._data = camera_data or CameraData(
+            (_pos[0], _pos[1], 0.0),
+            (_up[0], _up[1], 0.0),
+            (0.0, 0.0, -1.0),
+            zoom or 1.0
+        )
+
+        _proj = projection or (
+            -self._window.width/2, self._window.width/2,
+            -self._window.height/2, self._window.height/2
+        )
+        self._projection = projection_data or OrthographicProjectionData(
+            _proj[0], _proj[1],  # Left and Right.
+            _proj[2], _proj[3],  # Bottom and Top.
+            near or 0.0, far or 100.0,  # Near and Far.
+
+            viewport or (0, 0, self._window.width, self._window.height)  # Viewport
+        )
+
+        self._ortho_projector: OrthographicProjector = OrthographicProjector(
+            window=self._window,
+            view=self._data,
+            projection=self._projection
+        )
+
+    @property
+    def view_data(self) -> CameraData:
+        """
+        Return the view data for the camera. This includes the
+        viewport, position, forward vector, up direction, and zoom.
```

Comment:
I think `CameraData` just contains `position`, `up`, `forward` and `zoom` and **not** the `viewport`, correct?

Reply:
A print of the view_data property shows:
```
CameraData(position=(0.0, 0.0, 0.0), up=(0.0, 1.0, 0.0), forward=(0.0, 0.0, -1.0), zoom=1.0)
```

Reply:
yup this doc string was written before a change in the data structures good call